jupyterhub/tests/test_api.py

```diff
@@ -1549,6 +1549,92 @@ def normalize_token(token):
     }
 
 
+# ------------------
+# Activity API tests
+# ------------------
+
+
+async def test_update_activity_403(app, user, admin_user):
+    token = user.new_api_token()
+    r = await api_request(
+        app,
+        "users/{}/activity".format(admin_user.name),
+        headers={"Authorization": "token {}".format(token)},
+        data="{}",
+        method="post",
+    )
+    assert r.status_code == 403
+
+
+async def test_update_activity_admin(app, user, admin_user):
+    token = admin_user.new_api_token()
+    r = await api_request(
+        app,
+        "users/{}/activity".format(user.name),
+        headers={"Authorization": "token {}".format(token)},
+        data=json.dumps({"last_activity": utcnow().isoformat()}),
+        method="post",
+    )
+    r.raise_for_status()
+
+
+@mark.parametrize(
+    "server_name, fresh",
+    [
+        ("", True),
+        ("", False),
+        ("exists", True),
+        ("exists", False),
+        ("nope", True),
+        ("nope", False),
+    ],
+)
+async def test_update_server_activity(app, user, server_name, fresh):
+    token = user.new_api_token()
+    sp = user.spawners["exists"]
```

Comment:
@minrk, how is `sp` being used? A flag for spawner present? I'm not seeing it used in test.

Reply:
Updated to remove this since it's not actually needed.

Thanks for the review, @willingc!